packages/pilot/src/pages/Anticipation/Anticipation.js

```diff
@@ -0,0 +1,911 @@
+
+    return updateBulk(client, {
+      automaticTransfer: isAutomaticTransfer,
+      bulkId,
+      paymentDate,
+      recipientId,
+      requestedAmount: value || requestedAmount,
+      timeframe,
+    }).then((bulk) => {
+      this.setState({
+        loading: false,
+        requestedAmount,
+      })
+
+      return bulk
+    })
+  }
+
+  confirmAnticipation () {
+    const {
+      bulkId,
+      recipient: {
+        id: recipientId,
+      },
+    } = this.state
+
+    const {
+      client,
+      t,
+    } = this.props
+
+    return confirmBulk(client, {
+      bulkId,
+      recipientId,
+    })
+      .then(({ status }) => {
+        this.setState({
+          bulkAnticipationStatus: status,
+          currentStep: 'result',
+          loading: false,
+          statusMessage: t('pages.anticipation.anticipation_success'),
+          stepsStatus: getStepsStatus('result', 'success'),
+        })
+      })
+      .catch(pipe(getErrorMessage, error => this.setState({
+        currentStep: 'result',
+        loading: false,
+        statusMessage: error,
+        stepsStatus: getStepsStatus('result', 'error'),
+      })))
+  }
+
+  createAnticipation (value) {
+    const {
+      isAutomaticTransfer,
+      paymentDate,
+      recipient: {
+        id: recipientId,
+      },
+      requestedAmount,
+      timeframe,
+    } = this.state
+
+    const {
+      client,
+      limits: {
+        max,
+      },
+    } = this.props
+
+    return createBulk(client, {
+      automaticTransfer: isAutomaticTransfer,
+      paymentDate,
+      recipientId,
+      requestedAmount: value || requestedAmount,
+      timeframe,
+    })
+      .then(({
+        amount,
+        anticipation_fee: anticipationFee,
+        fee,
+        fraud_coverage_fee: fraudCovarageFee,
+        id,
+        status,
+      }) => {
+        this.setState({
+          approximateRequested: amount,
+          bulkAnticipationStatus: status,
+          bulkId: id,
+          error: null,
+          feesValues: {
+            anticipation: anticipationFee,
+            fraud: fraudCovarageFee,
+            otherFee: fee,
+          },
+          loading: false,
+          requestedAmount: max,
+        })
+      })
+  }
+
+  render () {
+    const {
+      approximateRequested,
+      businessCalendarError,
+      calendar,
+      currentStep,
+      error,
+      feesValues: {
+        anticipation,
+        fraud,
+        otherFee,
+      },
+      isAutomaticTransfer,
+      loading,
+      paymentDate,
+      recipient,
+      requestedAmount,
+      statusMessage,
+      stepsStatus,
+      timeframe,
+      transferCost,
+    } = this.state
+
+    const {
+      client,
+      error: limitsError,
+      limits: {
+        max,
+        min,
+      },
+      loading: limitsLoading,
+      t,
+    } = this.props
+
+    const totalCost = -(anticipation + fraud + otherFee)
+    const amount = approximateRequested + totalCost + transferCost
```

Comment:
acredito que isso aqui ajuda a resolver o problema de `NaN` na inicialização da pagina 
```suggestion
    const totalCostAndTransfer = totalCost + transferCost
    const amount = approximateRequested
      ? approximateRequested + totalCostAndTransfer
      : totalCostAndTransfer
```

Reply:
Boa! Corrigi porém de outra forma, achei mais simples adicionar valores default na local state do component, assim `approximateRequested` e `transferCost` serão sempre números válidos, evitando o NaN

Reply:
percebi que com essa validação acabou quebrando o fluxo de calcular os limites pois essa validação aqui não triga https://github.com/pagarme/pilot/blob/88920b3d899901e92e1b60d4e8a0b7cbffa7e16c/packages/pilot/src/pages/Anticipation/Anticipation.js#L337

Reply:
Eita, verdade! Corrigi para manter da forma como tinha sugerido então